### PR TITLE
Feature/unit tests

### PR DESCRIPTION
**Pull request by sdat2**
Hi, 

I am adding a some unit tests that will run as a github action for a few different versions of python at each push.

Currently it only pip installs the package and creates the test data, but I will add a few more.

I've added settings in the `pytest.ini` file that allow code blocks using `>>>` in python docstrings to act both as unit tests and examples, which should hopefully reduce duplication.

### CODE REVIEW

**Comment by sdat2** [comment]
`python-3.10` not able to install rasterio as detailed in #71 .

**Comment by herbiebradley** [comment]
Feel free to unpin `rasterio` from the requirements, then run the GitHub actions again! I tested with Python 3.10 and then just pip installed the `requirements.txt` as normal and everything seems to work.

**Comment by sdat2** [comment]
Hi Herbie, That makes sense, I'll try unpinning the `rasterio` version. 

RE using the `requirements.txt` file, I guess that that isn't an ideal thing to do if we want to make sure our pip package is compatible? Correct me if I'm wrong, but if you `pip install geograph` from somewhere else, that wouldn't be used? If that's right, that would undermine people's ability to easily build their packages on top of ours.

**Comment by sdat2** [comment]
Ah ok, so everything seems to be ok now for linux/mac, but I should probably fix windows as well for completeness.

**Comment by herbiebradley** [comment]
> Hi Herbie, That makes sense, I'll try unpinning the `rasterio` version.
> 
> RE using the `requirements.txt` file, I guess that that isn't an ideal thing to do if we want to make sure our pip package is compatible? Correct me if I'm wrong, but if you `pip install geograph` from somewhere else, that wouldn't be used? If that's right, that would undermine people's ability to easily build their packages on top of ours.

Yes, I just meant that for dev purposes I use the `requirements.txt`. Of course the actual place to unpin is `setup.py`, as you have done. :)

**Comment by herbiebradley** [comment]
Regarding the GDAL Windows issue, you may be able to fix this by including a GDAL wheels download and install in the configuration somehow: https://stackoverflow.com/questions/55583234/installation-fails-for-fiona-and-geopandas-with-gdal-on-python-3-6-on-microsoft

But it's not a big deal for now if it doesn't work - interestingly this was so annoying for `geopandas` that they don't try to solve it for the user and instead suggest Windows users use `pyogrio` which is a fiona alternative.

**Review by herbiebradley — APPROVED** [review]
Looks good! I was wondering what is the reason for adding 'pipwin' and 'descartes' to 'setup.py'? You could also add 'matplotlib' and 'seaborn' since I noticed some of our code requires them and I couldn't figure out why we didn't include them, unless you can remember?

**Comment by sdat2** [comment]
Hi, sure, I'm very happy to give up on Windows. I guess we can direct any windows users to this stack exchange answer: https://stackoverflow.com/a/65082853 if they want to use this package.

File: .github/workflows/python-package.yml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+        python-version: ${{ matrix.python-version }}
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        python -m pip install flake8 pytest matplotlib seaborn 
+        pip install -e .
+    - name: Lint with flake8
+      run: |
+        # stop the build if there are Python syntax errors or undefined names
+        flake8 . --count --select=E9,F63,F7,F82 --show-source --statistics
```
[the remark this answers is not in the document]
Comment (sdat2) [reply .github/workflows/python-package.yml:36]: Hi, I'm not sure, I originally adapted this from a github actions example. Given that we have pylint anyway, I will get rid of this section.

**Comment by sdat2** [comment]
Hi Herbie, I only added  'pipwin' and 'descartes'  to the `setup.py` to try to remove the GDAL error based on some stack exchange answers. I will remove them again in the next commit.

**Review by herbiebradley — APPROVED** [review]

**Comment by sdat2** [comment]
(The CI branch was still triggered by `master` rather than `main`)

**Comment by sdat2** [comment]
I've added only one unit test, which is more of a "smoke test". It might be worth adding some more before we merge this branch in.

**Review by herbiebradley — APPROVED** [review]
Looks good! I don't think it's necessary to fix the pylint errors for this PR, I can either fix them or turn off the unnecessary error codes later.

**Comment by sdat2** [comment]
Sure, I can merge and then add some more unit tests in a future pull request.

**Pull request merged**